Hometask2.playground/Contents.swift

```diff
@@ -0,0 +1,178 @@
+//1. Универсальные функции сложения, вычитания, умножения и деления для всех типов чисел (перегрузка функций, одинаковые названия для функций, разные типы параметров)
+
+import Darwin
+
+//The first realization via generics
+sumTwoValues(4, plus: 6)
+subtractTwoValues(6.3, minus: 8.99)
+multiplyTwoValues(5.545645, by: 8.54364564)
+divideTwoValues(4.0, by: 2.2)
+
+func sumTwoValues<T: Numeric>(_ firstNumber: T, plus secondNumber: T) -> T {
+    return firstNumber + secondNumber
+}
+
+func subtractTwoValues<T: Numeric>(_ firstNumber: T, minus secondNumber: T) -> T {
+    return firstNumber - secondNumber
+}
+
+func multiplyTwoValues<T: Numeric>(_ firstNumber: T, by secondNumber: T) -> T {
+    return firstNumber * secondNumber
+}
+
+// Because of Numeric protocol limitation , for division using -> FloatingPoint
+func divideTwoValues<T: FloatingPoint>(_ firstNumber: T,  by secondNumber: T) -> T {
+    return firstNumber / secondNumber
+}
+
+//The second realization via functions overloading
+//Sum realization via overloading
+
+let theFirstSum: Int     = digitsSum(2, plus: 2)
+let theSecondSum: Double = digitsSum(4, plus: 8)
+let theThirdSum: Int     = digitsSum(5.0, plus: 8.0)
+let theFourthSum: Double = digitsSum(5.5, plus: 5.6)
```

Comment:
We don't use articles `a`, `an` and `the` in namings so you can drop them

Reply:
fixed